File: tests/0242_valid_anagram_test.cpp
#include "0242_valid_anagram.cpp"
#include <doctest.h>

TEST_CASE("isAnagram")
{
    Solution s = Solution();
    CHECK(s.isAnagram("rat", "tar"));
    CHECK(s.isAnagram("anagram", "nagaram"));
    CHECK_FALSE(s.isAnagram("ab", "a"));
    CHECK_FALSE(s.isAnagram("ac", "bb"));
    CHECK_FALSE(s.isAnagram("rat", "car"));
}

File: tests/0242_valid_anagram_test.py
import importlib

module = importlib.import_module("algorithms.0242_valid_anagram")


def test_is_anagram():
    s = module.Solution()
    assert s.isAnagram("rat", "tar")
    assert s.isAnagram("anagram", "nagaram")
    assert not s.isAnagram("ab", "a")
    assert not s.isAnagram("ac", "bb")
    assert not s.isAnagram("rat", "car")
